instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/r2dbc/R2dbcInstrumentingPostProcessor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.r2dbc;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.internal.EmbeddedInstrumentationProperties;
+import io.opentelemetry.instrumentation.r2dbc.v1_0.R2dbcTelemetry;
+import io.r2dbc.spi.ConnectionFactory;
+import io.r2dbc.spi.ConnectionFactoryOptions;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.r2dbc.OptionsCapableConnectionFactory;
+
+class R2dbcInstrumentingPostProcessor implements BeanPostProcessor {
+
+  private final ObjectProvider<OpenTelemetry> openTelemetryProvider;
+  private final boolean statementSanitizationEnabled;
+
+  R2dbcInstrumentingPostProcessor(
+      ObjectProvider<OpenTelemetry> openTelemetryProvider, boolean statementSanitizationEnabled) {
+    this.openTelemetryProvider = openTelemetryProvider;
+    this.statementSanitizationEnabled = statementSanitizationEnabled;
+  }
+
+  @Override
+  public Object postProcessAfterInitialization(Object bean, String beanName) {
+    if (!(bean instanceof ConnectionFactory)) {
+      return bean;
+    }
+    ConnectionFactory connectionFactory = (ConnectionFactory) bean;
```

Comment:
Perhaps in this case it may be worth to do something like https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/293d9d8ff78468935654d852b85272ecec1503aa/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/jdbc/DataSourcePostProcessor.java#L28 to avoid potential double wrapping